### PR TITLE
DEV-2030 Refactor v2/awards/

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
Due to it being a blocker to a number of Frontend tickets, #1644 was rushed through the review process and merged it without even correct documentation, tests or complete review with frontend representation. This PR is the "second half" of work for DEV-1801

**Technical details:**
During the work, it was decided to refactor the data layers and remove the serializers

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
	- [ ] Backend
	- [ ] Frontend
4. [x] Matview impact assessment completed (N/A)
5. [ ] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket [DEV-2030](https://federal-spending-transparency.atlassian.net/browse/DEV-2030):
	- [x] Link to this Pull-Request
	- [x] Performance evaluation of affected (API | Script | Download)
	- [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Comment by kbard** [comment]
This is more of a question than a comment.  I don't see `generated_unique_award_id`, `id`, `total_obligation`, `base_and_all_options_value`, or `base_exercised_options_val` listed [in the contract](https://github.com/fedspendingtransparency/usaspending-website/blob/dev/api_contracts/contracts/award/AwardProfile.md#idvresponse-object) yet I'm pretty sure they're supposed to be returned by the API.  Why for so?  Is there a newer contract I'm not seeing or are they just implied or was there a verbal agreement that's not reflected in the contract?

File: usaspending_api/awards/v2/data_layer/orm_mappers.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        ("other_than_full_and_o_desc", "other_than_full_and_o_desc"),
+        ("type_set_aside_description", "type_set_aside_description"),
+        ("commercial_item_acquisitio", "commercial_item_acquisitio"),
+        ("commercial_item_test_desc", "commercial_item_test_desc"),
+        ("evaluated_preference_desc", "evaluated_preference_desc"),
+        ("fed_biz_opps_description", "fed_biz_opps_description"),
+        ("small_business_competitive", "small_business_competitive"),
+        ("fair_opportunity_limi_desc", "fair_opportunity_limi_desc"),
+        ("product_or_service_code", "product_or_service_code"),
+        ("product_or_service_co_desc", "product_or_service_co_desc"),
```
Comment (kbard) [inline usaspending_api/awards/v2/data_layer/orm_mappers.py]: I believe this is being referred to as `product_or_service_desc` in the contract.  Perhaps the contract needs to be updated?

File: usaspending_api/awards/v2/data_layer/orm_mappers.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        ("naics", "naics"),
```
Comment (kbard) [inline usaspending_api/awards/v2/data_layer/orm_mappers.py]: There is also a `naics_description` in the contract.

**Comment by tony-sappe** [comment]
> This is more of a question than a comment. I don't see `generated_unique_award_id`, `id`, `total_obligation`, `base_and_all_options_value`, or `base_exercised_options_val` listed [in the contract](https://github.com/fedspendingtransparency/usaspending-website/blob/dev/api_contracts/contracts/award/AwardProfile.md#idvresponse-object) yet they're being returned by the API (and I'm pretty sure they're supposed to be). Why for so? Is there a newer contract I'm not seeing or are they just implied or was there a verbal agreement that's not reflected in the contract?

There are changes in an open [usaspending-website PR](/api_contracts/contracts/award/AwardProfile.md)
My changes don't match the contract 100% yet, I added at least 1-2 additional fields and I had some questions of pre-existing fields not in the API contract and didn't know if they should remain

File: usaspending_api/awards/v2/data_layer/orm_mappers.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/awards/v2/data_layer/orm_mappers.py]: Or the field should be renamed to match the contract

Comment (tony-sappe) [reply usaspending_api/awards/v2/data_layer/orm_mappers.py]: API contract in PR does use `product_or_service_co_desc`

**Comment by tony-sappe** [comment]
Don't merge into dev until Sprint 74 changes are in staging!

File: usaspending_api/awards/tests/test_awards_v2.py [hunk 3]
```diff
@@ -151,92 +195,92 @@ def awards_and_transactions(db):
...
 
-    resp = client.get('/api/v2/awards/1/')
+    resp = client.get("/api/v2/awards/1/")
     assert resp.status_code == status.HTTP_200_OK
     assert json.loads(resp.content.decode("utf-8")) == expected_response_asst
 
-    resp = client.get('/api/v2/awards/2/')
+    resp = client.get("/api/v2/awards/2/")
     assert resp.status_code == status.HTTP_200_OK
     assert json.loads(resp.content.decode("utf-8")) == expected_response_cont
```
Comment (IsaacRay) [inline usaspending_api/awards/tests/test_awards_v2.py]: You don't test for an IDVResponse with any of these. The award models that you mommy-make have categories of "contract", and "grant", so you never trigger the construct_idv_response functionality.

Comment (tony-sappe) [reply usaspending_api/awards/tests/test_awards_v2.py]: @IsaacRay What about `test_awards_idv_v2.py`?

**Pull request merged**